```rust
error[E0308]: mismatched types
     --> /Users/servo/buildbot/slave/mac-dev-unit/build/target/geckolib/debug/build/style-8e72fa08530fa207/out/properties.rs:19775:49
      |
19775 |                 if let Some(color) = color_name(ident) {
      |                                                 ^^^^^ expected &str, found struct `cssparser::CompactCowStr`
      |
      = note: expected type `&str`
                 found type `cssparser::CompactCowStr<'_>`
      = help: here are some functions which might fulfill your needs:
              - .trim()
              - .trim_left()
              - .trim_right()

error[E0308]: mismatched types
     --> /Users/servo/buildbot/slave/mac-dev-unit/build/target/geckolib/debug/build/style-8e72fa08530fa207/out/properties.rs:19776:24
      |
19776 |                     Ok(color)
      |                        ^^^^^ expected enum `gecko_bindings::structs::root::mozilla::LookAndFeel_ColorID`, found reference
      |
      = note: expected type `gecko_bindings::structs::root::mozilla::LookAndFeel_ColorID`
                 found type `&gecko_bindings::structs::root::mozilla::LookAndFeel_ColorID`

error: aborting due to 2 previous errors

error: Could not compile `style`.
```

Try `color_name(&ident)` and `Ok(*color)`, and run `./mach check-geckolib` to make sure it compiles.

--- a/components/style/properties/longhand/color.mako.rs
+++ b/components/style/properties/longhand/color.mako.rs
@@ -121,21 +121,20 @@
 
         impl SystemColor {
             pub fn parse<'i, 't>(input: &mut Parser<'i, 't>) -> Result<Self, ParseError<'i>> {
-                #[cfg(feature = "gecko")]
-                use std::ascii::AsciiExt;
-                static PARSE_ARRAY: &'static [(&'static str, SystemColor); ${len(system_colors)}] = &[
-                    % for color in system_colors:
-                        ("${color}", LookAndFeel_ColorID::eColorID_${to_rust_ident(color)}),
-                    % endfor
-                ];
-
-                let ident = input.expect_ident()?;
-                for &(name, color) in PARSE_ARRAY.iter() {
-                    if name.eq_ignore_ascii_case(&ident) {
-                        return Ok(color)
+                ascii_case_insensitive_phf_map! {
+                    color_name -> SystemColor = {
+                        % for color in system_colors:
+                            "${color}" => LookAndFeel_ColorID::eColorID_${to_rust_ident(color)},
+                        % endfor
                     }
                 }
-                Err(SelectorParseError::UnexpectedIdent(ident).into())
+
+                let ident = input.expect_ident()?;
+                if let Some(color) = color_name(&ident) {
+                    Ok(*color)
+                } else {
+                    Err(SelectorParseError::UnexpectedIdent(ident).into())
+                }
             }
         }
     % endif

